### DISCUSSION TITLE
Comparision graph of PBE, OS-DMFT, CSC-DMFT

### DISCUSSION THREAD

**Discussion by TommyHenz** [TRIQS/solid_dmft#32] [Q&A] [answered]
Hi, I'm just writing to ask if it's possible to have the script to plot the top graph shown in [this CSC tutorial](https://triqs.github.io/solid_dmft/tutorials/Ce2O3_csc_w90/tutorial.html). I've done my PBE, ON-DMFT, CSC-DMFT calculations, but couldn't understand exactly how to plot the graph shown on the top of that tutorial. Since I've got limited knowledge with Python, was hoping to get a help from you guys.

**Comment by the-hampel** [comment, marked as answer]
Hi @TommyHenz,

Did you already calculate the total energy for various strains with solid_dmft and want to plot it? 

Otherwise there is maybe some misunderstanding here. The plot shown in the top of the tutorial will not be reproduced by the tutorial, it only shows how to perform a single calculation of the plot, which in total would consist of various total energy calculations (each of them represented by a marker). 

During the tutorial we only calculate the total energy of a single structure at 0% strain. To create the plot you would have to change the lattice constant in your input structure and rerun the whole DFT+DMFT calculation for each strain you would like to calculate. 

For the one-shot calculations you also have to manually add the DFT energy to obtain the final energy to the value given in the solid_dmft output. That is, because in one-shot calculations solid_dmft does not know the DFT total energy, and sets its value to 0.0. 

Best,
Alex

**Reply by TommyHenz** [reply]
Many thanks, I understand. I'm able to run the calculations, it will take some time to complete as a whole, but that's okay.
For example for OS calculation, I checked the dmft output file and for a iteration is says "total energy = -70". This means that from my DFT energy of 5000 eV, I just substract this right?

Just a quick question in regarding that tutorial. It says a few times that you compare with a "reference data" but it doesn't mention which reference data you mean. Could you let me know what calculation was run for the reference data?

Greetings,
Tommy

**Comment by phibeck** [comment]
Hi @TommyHenz ,

as we write in the tutorial, the parameters given there are simplified for demonstration purposes. The reference data contained in the directory `solid_dmft/doc/tutorials/Ce2O3_csc_w90/ref/` contains a set of data for converged parameters. For this you will have to do the usual procedure: check convergence with number of k-points and energy cutoff for DFT, make sure your Wannier functions are a good fit and converge the important parameters for HubbardI in DMFT, which are mostly the mesh sizes of your Greens functions, i.e. number of Legendre polynomials, Matsubara frequencies and Tau points. This is a good exercise to understand the parameters when switching to a more complicated material.

Regarding the energies: for DFT the plot simply shows the total energy as usual, for OS-DMFT it shows the same DFT energy plus the total energy from DMFT (`E_tot` in the observables file), and for CSC-DMFT it is just `E_tot`, since the DFT energy is already included in that case. Be aware that all energies in solid_dmft are given in eV, while QE prints them in Ry, so you have to do the conversion to have everything in the same unit. Note also that there was a small bug reading the correct DFT energy for CSC calculations in solid_dmft that was fixed a few months ago and will shift the minimum position for CSC-DMFT slightly if you compare your results to the figure on the webpage.

Best,
Sophie
